decoder.go

```diff
@@ -236,12 +240,26 @@ func newDecoder(codecID C.enum_AVCodecID, avFramePool *framePool, logger logging
 	// output JPEG color range frames YUVJ420P.
 	codecCtx.pix_fmt = C.AV_PIX_FMT_YUV420P
 
+	// Set extradata if provided
+	if len(extraData) > 0 {
+		codecCtx.extradata_size = C.int(len(extraData))
+		codecCtx.extradata = (*C.uint8_t)(C.av_malloc(C.size_t(codecCtx.extradata_size)))
+		if codecCtx.extradata == nil {
+			C.avcodec_close(codecCtx)
+			return nil, errors.New("av_malloc() failed for extradata")
+		}
+		C.memcpy(unsafe.Pointer(codecCtx.extradata), unsafe.Pointer(&extraData[0]), C.size_t(codecCtx.extradata_size))
```

Comment:
Nicely done -- did not realize this how you set the global headers for MPEG4.

Reply:
I didn't either! I tried to make h264/5 also use this global headers flow, but I think NAL units work differently i.e. every frame